src/foam/nanos/fs/Storage.java

```diff
@@ -32,6 +31,59 @@ public Storage (String root, boolean isResource) {
     resourceDir_ = root;
   }
 
+  protected FileSystem getFS() {
+    FileSystem fs = null;
+
+    if ( isResource() ) {
+      String resourceJar = System.getenv("RES_JAR_HOME");
+      if (!SafetyUtil.isEmpty(resourceJar)) {
+        Path resourceJarPath = Paths.get(resourceJar);
+        try {
+          URI resourceJarURI = new URI("jar", resourceJarPath.toUri().toString(), null);
+          try {
+            fs = FileSystems.getFileSystem(resourceJarURI);
+            return fs;
+          } catch (FileSystemNotFoundException e) {
+            Map<String, String> env = new HashMap<>();
+            env.put("create", "true");
+            fs = FileSystems.newFileSystem(resourceJarURI, env);
+            return fs;
+          }
+        } catch (URISyntaxException | IOException e) {
+          throw new RuntimeException(e);
+        }
+      } else {
+        throw new RuntimeException("Couldn't open resource FS, missing jar " + resourceJar);
+      }
+    } else {
+      fs = FileSystems.getDefault();
+    }
+
+    return fs;
+  }
+
+  public Path getPath(String name) {
+    FileSystem fs = getFS();
+    if ( fs == null ) return null;
+
+    Path path;
+    if ( isResource() ) {
```

Comment:
This class is getting really complicated, I think we should split it up into multiple implementations of the same interface.  One implementation for a regular directory and another for reading from resources.

I also don't think we should be returning Path objects, but instead we should abstract the caller away from paths and instead provide methods to get an iStream or to read all bytes of a file.